sim/core/spell_outcome.go

```diff
@@ -538,13 +532,10 @@ func (spell *Spell) OutcomeRangedCritOnlyNoHitCounter(sim *Simulation, result *S
 func (spell *Spell) outcomeRangedCritOnly(sim *Simulation, result *SpellResult, attackTable *AttackTable, countHits bool) {
 	// Block already checks for this, but we can skip the RNG roll which is expensive.
 	if spell.Unit.PseudoStats.InFrontOfTarget {
-		roll := sim.RandomFloat("White Hit Table")
-		chance := 0.0
-
 		if result.applyAttackTableCritSeparateRoll(sim, spell, attackTable, countHits) {
-			result.applyAttackTableBlock(spell, attackTable, roll, &chance)
+			result.applyAttackTableBlock(sim, spell, attackTable)
 		} else {
-			if !result.applyAttackTableBlock(spell, attackTable, roll, &chance) {
+			if !result.applyAttackTableBlock(sim, spell, attackTable) {
 				result.applyAttackTableHit(spell, countHits)
 			}
 		}
```

Comment:
Remove ranged Block outcomes here for consistency if you're removing them everywhere else, which also means that `InFrontOfTarget` should no longer matter.